### PR TITLE
Update, Delete, Aggregate

### PR DESCRIPTION
**Pull request by Nelofarzabi**
268995721-76d67bdb-07a5-451d-9fe8-d4478f33538f
268995751-af90523e-274e-45cd-985e-a222d4399aeb
268995826-ce7180e9-6728-40e0-93c1-7d8a55f35dd6
268995884-1a233af1-9fc9-41e8-8294-18a6d2315344
268995895-a349d221-47e7-47af-b017-da46118e07c8
268995898-e7f1e1ad-e433-4bad-95b7-20fa71422633
268995910-e91e9fe3-d0da-4933-9988-e63bd23100df
268995921-020eaa37-42f3-49f7-8ebb-26c8dc5f66a0
268995933-f43b4af8-9ce7-4840-8373-4c8cf25aa744
268995947-1306b5eb-5fe5-486a-94ac-08fb3eb42f3d
268995976-1779b886-622f-45c0-9a3e-6c30754fdc0b
268995987-81053e7f-cbac-431e-bd16-fc0c5fb7a3a1
268995993-9a7120b7-dc7c-427d-bf56-d9f9b9c04d4b
268996008-40ca0534-8658-445e-bddb-f98c7ed8ac37
268996018-9b8fdbb4-5aef-4577-9d67-45cf676a9476
268996028-201b0741-8484-4c66-b2ae-c43c3dca85d2
268996041-5b6fc76f-49bb-4716-b74e-98179560fe9b
268996050-488b657f-e184-493d-8693-a5da3f547cdb
268996061-c22b50c7-3464-4699-a491-4c822a348d22
268996068-6fa44090-6748-4726-9144-1d9739de1300
268996074-1ccbf144-bd45-42b1-b12b-9050e74a890a
268996074-1ccbf144-bd45-42b1-b12b-9050e74a890a
268996078-8be3be66-a359-47d6-957c-bb416e0ce1c8
268996082-0f3cd997-e0a0-4cea-b5b2-a2e03fda1d88
268996089-b65b5752-27b8-451e-979a-48940bba8982
268996098-3a6e565c-c1bb-4de9-8bd1-20d9e2f2b35f
268996108-ff777808-c724-44ea-b1ea-bb31d8b058e6

In this PR, the folowing were affected;

- Added a column species of type string to your animals table. Modify your schema.sql file.
- Inserted new sets of data
- Created TRANSACTIONS to update and delete from table
- Wrote queries for aggregate functions

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -16,8 +16,115 @@ SELECT * FROM animals WHERE name != 'Gabumon'; --Find all animals not named Gabu
...
+Update the animals table by setting the 
+species column to digimon for all 
+animals that have a name ending in mon.
+*/
+
+BEGIN;
+
+UPDATE animals
+SET species = 'digimon'
+WHERE name LIKE '%mon';
```
Comment (topeogunleye) [inline queries.sql:43]: Hello! 👋. You've done a good job in utilizing SQL transactions, but there's a small gap to fully meet the project requirements. 📝

#### Highlights:
- You've correctly initiated a transaction using `BEGIN`.
- Your `UPDATE` query for setting the `species` column to 'unspecified' is spot on! 👏

#### Areas for Improvement:
- According to the project requirements, you are explicitly asked to include steps to verify that the change was made and then rolled back. Your code lacks the `SELECT` queries that confirm these steps.

Here's how you could improve it:

```sql
BEGIN; -- start transaction

UPDATE animals
SET species = 'unspecified'; -- make change

SELECT species from animals; -- 👈 Add this line to verify that change was made

ROLLBACK; -- roll back the change

SELECT species from animals; -- 👈 Add this line to verify that change was undone
```

Please include these additional `SELECT` queries to fully meet the project requirements. Looking forward to your next commit! 🚀

Best,
Temitope

File: queries.sql [hunk 2]
```diff
@@ -16,8 +16,115 @@ SELECT * FROM animals WHERE name != 'Gabumon'; --Find all animals not named Gabu
...
+Inside a transaction update the animals table 
+by setting the species column to unspecified. 
+Verify that change was made. Then roll back 
+the change and verify that the species 
+columns went back to 
+the state before the transaction.
+*/
 BEGIN;
 UPDATE animals
 SET species = 'unspecified';
```
Comment (topeogunleye) [inline queries.sql]: Hello! 👋 Thank you for submitting your code for review. You've done a good job in utilizing SQL transactions, but there's a small gap to fully meet the project requirements. 📝

#### Highlights:
- You've correctly initiated a transaction using `BEGIN`.
- Your `UPDATE` query for setting the `species` column to 'unspecified' is spot on! 👏

#### Areas for Improvement:
- According to the project requirements, you are explicitly asked to include steps to verify that the change was made and then rolled back. Your code lacks the `SELECT` queries that confirm these steps.

Here's how you could improve it:

```sql
BEGIN; -- start transaction

UPDATE animals
SET species = 'unspecified'; -- make change

SELECT species from animals; -- 👈 Add this line to verify that change was made

ROLLBACK; -- roll back the change

SELECT species from animals; -- 👈 Add this line to verify that change was undone
```

Please include these additional `SELECT` queries to fully meet the project requirements. Looking forward to your next commit! 🚀

Best,
Temitope

File: queries.sql [hunk 3]
```diff
@@ -16,8 +16,115 @@ SELECT * FROM animals WHERE name != 'Gabumon'; --Find all animals not named Gabu
...
+COMMIT;
+
+/*
+Inside a transaction delete all records in the animals table, 
+then roll back the transaction.
+*/
+BEGIN;
+
+DELETE FROM animals;
+ROLLBACK;
```
Comment (topeogunleye) [inline queries.sql]: Hello! . You're on the right track with using transactions and the `DELETE` statement. However, there are some minor adjustments needed to fully meet the project requirements. 🛠️

#### Highlights:
- You've initiated a transaction correctly with `BEGIN;`. 🎉
- Your `DELETE FROM animals;` query is spot on for removing all records from the `animals` table. 👍
- Good use of `ROLLBACK;` to undo the deletion operation. 👏

#### Areas for Improvement:
- Verification Missing: The project requirements specify that you should include `SELECT COUNT(*) FROM ANIMALS;` queries before and after the `ROLLBACK;` to verify that records are actually deleted and then restored. These lines are missing in your code.

Here's how you could complete the requirements:

```sql
BEGIN; -- start transaction

DELETE FROM animals; -- Delete all records

SELECT COUNT(*) FROM animals; -- 👈 Add this line to verify that records were deleted

ROLLBACK; -- Roll back the transaction

SELECT COUNT(*) FROM animals; -- 👈 Add this line to verify that records were restored
```

Please make these adjustments and re-submit your code. Looking forward to your next commit! 🚀

Best,
Temitope

File: queries.sql [hunk 4]
```diff
@@ -16,8 +16,115 @@ SELECT * FROM animals WHERE name != 'Gabumon'; --Find all animals not named Gabu
...
+SELECT neutered, AVG(escape_attempts) AS AVG_OF_ESCAPE
+FROM animals
+WHERE neutered = TRUE AND date_of_birth BETWEEN '1990-01-01' AND '2000-12-31'
+GROUP BY neutered;
+
+-- Not neutered the average number of escape attempts
+SELECT neutered as NOT_NEUTERED, AVG(escape_attempts) AS AVG_OF_ESCAPE
+FROM animals
+WHERE neutered = FALSE AND date_of_birth BETWEEN '1990-01-01' AND '2000-12-31'
+GROUP BY neutered;
```
Comment (topeogunleye) [inline queries.sql]: Hello! 👋. You've got some interesting calculations in there, and it's clear you have a good grasp of using `AVG()` and `GROUP BY`. However, your current code does not meet the specific project requirements. 🛠️

#### Highlights:
- Good use of the `AVG()` function to calculate average values. 🌟
- Your `GROUP BY` and `WHERE` clauses are well-structured. 👍

#### Areas for Improvement:
1. **Wrong Criteria**: The project asks for the average number of escape attempts **per animal type** for animals born between 1990 and 2000. You've calculated it based on whether the animals are neutered or not, which is not what was required.

Here's how your code can meet the requirements:

```sql
SELECT species, AVG(escape_attempts) AS AVG_OF_ESCAPE
FROM animals
WHERE date_of_birth BETWEEN '1990-01-01' AND '2000-12-31'
GROUP BY species;
```

Please update your query to align with the project requirements and resubmit your code. Can't wait to see your updated work! 🚀

Best,
Temitope

**Review by topeogunleye — CHANGES_REQUESTED** [review]
Hi @Nelofarzabi, 

Good job so far!,
Hey there! I've just gone through your SQL project, and I'm really impressed by what I see. You've nailed several key elements of database management, and it's clear that you have a solid understanding of SQL syntax and functionality. 🤩🚀

Try-Again

> ## (Highlights) Good Points: 💪 
### Outstanding Work on the SQL Project! 🌟👏

#### Phenomenal Highlights:

1. **File Organization**: You have meticulously followed the project guidelines by organizing your SQL code into separate files for schema, data, and queries. This shows not just your technical skills, but also your ability to read and adhere to specifications. 📝✅
  
2. **Schema Design**: Your ALTER TABLE command to add a new species column with VARCHAR(255) is spot-on. You've demonstrated great understanding of how to modify database schemas. 🛠️💪

3. **Data Insertions**: The way you populated the `animals` table is flawless. Your INSERT INTO statements are well-formed, and the sample data you used is not only appropriate but also quite imaginative! This will help with data testing later on. 🐾🎉
  
4. **Attention to Detail**: Your usage of different data types, such as strings for names and booleans for the `neutered` field, show a deep understanding of how data should be structured. This is foundational for database integrity and query performance. 👀🌟

5. **Data Diversity**: I love how you varied the data for escape attempts, neutered status, and weights. This attention to detail makes your sample database rich for querying and shows that you're thinking like a real-world developer. 🌍👨‍💻

6. **Great Commenting**: Although SQL commands are relatively straightforward, the neatness of your code makes it even easier to understand. And when code is easy to understand, it's easier to maintain. 📚🔍

Your work here is an exemplary demonstration of how to properly manage databases, from modifying schema to data insertion. I am thrilled by your submission and can't wait to see what you come up with next! 🌠🔥

Keep up the amazing work!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- Nothing to mention.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @topeogunleye in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by roshan-bajgain — APPROVED** [review]
Hi @Nelofarzabi, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- GitHub flow is followed. ✔️
- Professional README file. ✔️
- Good job on making changes ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**